File: content/snippets/go/interview_lcci/17/04_missing_number.go
package main

import "fmt"

func missingNumber(nums []int) int {
	ans := 0
	for i, x := range nums {
		ans ^= (i + 1) ^ x
	}
	return ans
}

func main() {
	fmt.Println(missingNumber([]int{3, 0, 1}))                   // 2
	fmt.Println(missingNumber([]int{9, 6, 4, 2, 3, 5, 7, 0, 1})) // 8
}

File: content/snippets/typescript/interview_lcci/17/04_missing_number.ts
function missingNumber(nums: number[]): number {
    let ans = 0;
    for (let i = 0; i < nums.length; i++) {
        ans ^= (i + 1) ^ nums[i];
    }
    return ans;
}

console.log(missingNumber([3, 0, 1]));                   // 2
console.log(missingNumber([9, 6, 4, 2, 3, 5, 7, 0, 1])); // 8
